pkg/watcher/results/results.go

```diff
@@ -116,6 +120,31 @@ func (c *Client) ensureResult(ctx context.Context, o Object, opts ...grpc.CallOp
 		}
 	}
 
+	// Set the Result.Annotations and Result.Summary.Annotations fields if
+	// the object in question contains the required annotations.
+
+	if value, found := o.GetAnnotations()[annotation.ResultAnnotations]; found {
+		if annotations, err := parseAnnotations(annotation.ResultAnnotations, value); err != nil {
+			return nil, err
+		} else {
+			if res.Annotations == nil {
+				res.Annotations = make(map[string]string, len(annotations))
+			}
+			copyKeys(annotations, res.Annotations)
```

Comment:
Isn't res.Annotations empty? You should use curr.Annotations

Reply:
If we don't want to employ merge, was there any issue with the previous approach of just copying?

We can just delegate the responsibility of ensuring that the merge happens to integrators.

Reply:
Merging values made the code trickier, but ultimately, it seems a good idea thinking about future extensibility.